### PR TITLE
Delete unused things in OrbitProcess OrbitModule OrbitFunction

### PR DESCRIPTION
**Pull request by antonrohr**
This mainly deletes things that are not used anymore.

In OrbitFunction there are a few things moved around to be able to remove the enum OrbitType.

### CODE REVIEW

File: OrbitCore/CMakeLists.txt [hunk 1]
```diff
@@ -74,6 +74,7 @@ target_sources(
           OrbitThread.cpp
           Params.cpp
           Path.cpp
+          Pdb.cpp
```
Comment (dimitry-) [inline OrbitCore/CMakeLists.txt:77]: I take it the end goal is to remove this completely.

**Comment by ronaldfw** [comment]
There are some parts that were recently added for manual instrumentation. The changes look correct to me, but I'm not sure if there might be something that will be needed for manual instrumentation (which is not done yet) that is getting deleted here. @pierricgimmig should take a look before this is merged.

File: OrbitCore/CMakeLists.txt (hunk 1 above)
Comment (antonrohr) [reply OrbitCore/CMakeLists.txt:77]: Yes

**Comment by antonrohr** [comment]
> There are some parts that were recently added for manual instrumentation. The changes look correct to me, but I'm not sure if there might be something that will be needed for manual instrumentation (which is not done yet) that is getting deleted here. @pierricgimmig should take a look before this is merged.

I agree. I actually specifically left parts in that are used and tested it. From what I can tell by my testing this is what is currently happening:
If `Orbit.h` is used in an application profiled by Orbit, the `orbit_api::` functions (for example `orbit_api::Start("string")` are automatically hooked. That is all so far. 

We should think about where to put a manual instrumentation functionality, because I don't think OrbitFunction is the right place for that.

File: OrbitCore/OrbitFunction.cpp [hunk 2]
```diff
@@ -143,36 +145,17 @@ const char* Function::GetCallingConventionString() {
...
-      {"TrackUint64(", ORBIT_TRACK_UINT_64},
-      {"TrackFloat(", ORBIT_TRACK_FLOAT},
-      {"TrackDouble(", ORBIT_TRACK_DOUBLE},
-      {"TrackFloatAsInt(", ORBIT_TRACK_FLOAT_AS_INT},
-      {"TrackDoubleAsInt64(", ORBIT_TRACK_DOUBLE_AS_INT_64},
-  };
-  return function_name_to_type_map;
-}
-
-// Detect Orbit API functions by looking for special function names part of the
```
Comment (pierricgimmig) [inline OrbitCore/OrbitFunction.cpp:166]: Needed.

File: OrbitCore/OrbitModule.cpp [hunk 3]
```diff
@@ -97,137 +58,5 @@ void Module::LoadSymbols(const ModuleSymbols& module_symbols) {
...
-Pdb::Pdb(uint64_t module_address, uint64_t load_bias, std::string file_name,
-         std::string module_file_name)
-    : m_MainModule(module_address),
-      load_bias_(load_bias),
-      m_FileName(std::move(file_name)),
-      m_LoadedModuleName(std::move(module_file_name)) {
-  m_Name = Path::GetFileName(m_FileName);
-}
-
-//-----------------------------------------------------------------------------
```
Comment (pierricgimmig) [inline OrbitCore/OrbitModule.cpp:137]: This is needed

**Comment by antonrohr** [comment]
I reverted the things you mentioned back to how they were before.

File: OrbitCore/OrbitModule.cpp (hunk 3 above)
Comment (antonrohr) [reply OrbitCore/OrbitModule.cpp:137]: done

File: OrbitCore/OrbitFunction.cpp (hunk 2 above)
Comment (antonrohr) [reply OrbitCore/OrbitFunction.cpp]: done

File: OrbitCore/OrbitFunction.cpp [hunk 4]
```diff
@@ -143,36 +145,17 @@ const char* Function::GetCallingConventionString() {
 
   return kCallingConventions[calling_convention_];
 }
+bool Function::IsOrbitFunc() const {
+  static absl::flat_hash_set<const char*> markers{
+      "Start(",      "Stop(",        "StartAsync(",      "StopAsync(",
+      "TrackInt(",   "TrackInt64(",  "TrackUint(",       "TrackUint64(",
+      "TrackFloat(", "TrackDouble(", "TrackFloatAsInt(", "TrackDoubleAsInt64("};
 
-const absl::flat_hash_map<const char*, Function::OrbitType>&
```
[the remark this answers is not in the document]
Comment (antonrohr) [reply OrbitCore/OrbitFunction.cpp]: done

File: OrbitCore/OrbitFunction.h [hunk 5]
```diff
@@ -21,32 +21,6 @@ class Pdb;
 
 class Function {
  public:
-  enum OrbitType {
```
[the remark this answers is not in the document]
Comment (antonrohr) [reply OrbitCore/OrbitFunction.h:24]: done

**Comment by pierricgimmig** [comment]
> I reverted the things you mentioned back to how they were before.

cool, thanks!

**Pull request merged**